helpers/middleman/menu.rb

```diff
@@ -0,0 +1,266 @@
+load File::expand_path('./locale.rb', __dir__)
+load File::expand_path('./page.rb', __dir__)
+
+module MiddlemanMenuHelpers
+  include MiddlemanLocaleHelpers # for current_locale_obj method
+  include MiddlemanPageHelpers # for locale_page method
+
+
+  def get_menu_config(locale_obj: current_locale_obj)
```

Comment:
@irwalker @alto So this is what I wound up with for the menu configuration..  It's pretty fully tested; you can change icons and visibility and it might not break CI, but it's pretty thorough—both for the Helper and the DOM itself.

I could use some feedback on how to document it, or just a better way to rebuild it, but realistically a Static Ruby application with Middleman really isn't the same as a Rails application, so I can't see a better way to structure this.

Everything else is this repo is JSON, this is essentially just JSON with some helpers.  Middleman really doesn't give you a way to structure data outside of the `/data/file.json` very well.

Reply:
I think what is here is fine; what do you mean in terms of documentation?

Reply:
@kylorhall Now I understand why you asked me that question in the DM. Do I get that right, you want to document how to use `get_menu_config` and its parameters, like a `locale_obj` should look like?

Reply:
@alto @irwalker So, I don't actually want to document the parameters—I think that's pretty straightforward in this repo.

More like what the `_menu` data object is and could be…is that worth documenting?

```js
{
  // menu interface
  label: 'Features',
  rows: [
    {
      // row interface
      label: 'Features',
      links: [
        {
          // link interface
          title: 'Performance',
          href: '/performance'
        }
      ]
    }
  ]
}
```

There's a lot more optional and [mostly] required data in these interfaces than that too…

**Alternatively:**
 ❓ Are the examples in this file just sufficient, not worth documenting?
 ❓ tl;dr: Could you update this file/links without any documentation?

Reply:
I'm happy that the examples here are sufficiently self-documenting.